packages/contracts-bedrock/src/L2/CrossL2Inbox.sol

```diff
@@ -30,26 +23,111 @@ struct Identifier {
 /// @title CrossL2Inbox
 /// @notice The CrossL2Inbox is responsible for executing a cross chain message on the destination
 ///         chain. It is permissionless to execute a cross chain message on behalf of any user.
+/// @dev Processes cross-chain messages that are pre-declared in EIP-2930 access lists. Each message
+///      requires three specific access-list entries to be valid. It will verify that the storage
+///      slot containing the message checksum is "warm" (pre-accessed), which fails if not included
+///      in the tx's access list. Nodes pre-check message validity before execution. The checksum
+///      combines the message's `Identifier` and `msgHash` with type-3 bit masking.
 contract CrossL2Inbox is ISemver {
+    /// @notice Thrown when trying to execute a cross chain message on a deposit transaction.
+    error NoExecutingDeposits();
+
+    /// @notice Thrown when trying to validate a cross chain message with a checksum
+    ///         that is invalid or was not provided in the transaction's access list to set the slot
+    ///         as warm.
+    error NotInAccessList();
+
+    /// @notice Thrown when trying to validate a cross chain message with a block number
+    ///         that is greater than 2^64.
+    error BlockNumberTooHigh();
+
+    /// @notice Thrown when trying to validate a cross chain message with a timestamp
+    ///         that is greater than 2^64.
+    error TimestampTooHigh();
+
+    /// @notice Thrown when trying to validate a cross chain message with a log index
+    ///         that is greater than 2^32.
+    error LogIndexTooHigh();
+
     /// @notice Semantic version.
-    /// @custom:semver 1.0.0-beta.13
-    string public constant version = "1.0.0-beta.13";
+    /// @custom:semver 1.0.0-beta.14
+    string public constant version = "1.0.0-beta.14";
+
+    /// @notice The mask for the most significant bits of the checksum.
+    /// @dev    Used to set the most significant byte to zero.
+    bytes32 internal constant _MSB_MASK = bytes32(~uint256(0xff << 248));
```

Comment:
I would have made the comment to nit back due to runtime costs of this approach: https://github.com/ethereum-optimism/optimism/pull/14873#discussion_r1995924087

Fine with leaving it as it